cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -72,6 +73,12 @@ using internal::ToChars;
 namespace compute {
 namespace {
 
+bool kDefaultUseThreads = false;
```

Comment:
This is a bit confusing - it seems that we only use this default in segment aggregation in two places, and this is also not the default for hash aggregation? We can probably remove this variable and specify explicitly in segment aggregation